lib/puppet.rb

```diff
@@ -85,11 +86,6 @@ def self.[](param)
   # Load the base features.
   require 'puppet/feature/base'
 
-  # Store a new default value.
-  def self.define_settings(section, hash)
-    @@settings.define_settings(section, hash)
-  end
-
```

Comment:
I think it's ok to remove this method, but you may want to check rspec-puppet and puppetlabs_spec_helper to make sure they're not defining new settings as some kind of monkey patch backwards compatibility thing.

Reply:
A quick grep of rspec-puppet and puppetlabs_spec_helper doesn't show any `Puppet.define_settings` calls.

Reply:
Can we do a deprecation warning here instead? (when did I get more conservative than Josh!?!)

Something like, this is going away in Puppet 7, if you're using it you should call `Puppet.settings.define_settings` instead of `Puppet.settings`

Reply:
Haha, having  👶will do that. Next thing you know @justinstoller will be saying, "get off my lawn"

Reply:
Is `define_settings` something we actually want to be a part of the public api? I'm cool with the deprecation warning change; however, I thought that this was something of a "convenience" method within Puppet itself, not something we expected users to be executing on their own.

So, should this just say "Access to define_settings is going away in Puppet 7" or should it be like @justinstoller 's comment? I think I prefer the former, but I'm open to being outvoted...

Reply:
Fair, I'm fine with us not giving alternative ways to do this and having setting definition be a private api. I'm just concerned about old things that aren't explicitly private right now being currently used by _someone_.